File: Server/src/main/java/bgu/spl/net/api/bidi/EncodeDecode/NotificationPacket.java
package bgu.spl.net.api.bidi.EncodeDecode;

public class NotificationPacket extends Packet {

    boolean isPm;
    String postingUser;
    String content;

    public NotificationPacket(short opCode,  boolean isPm,String postingUser,String content) {
        super(opCode);
        this.isPm=isPm;
        this.postingUser=postingUser;
        this.content=content;
    }

    public boolean isPm() {
        return isPm;
    }

    public String getPostingUser() {
        return postingUser;
    }

    public String getContent() {
        return content;
    }
}

File: Client/src/NotificationPacket.cpp
//
// Created by alona on 29/12/2018.
//

#include "../include/NotificationPacket.h"

NotificationPacket::NotificationPacket(short op, char isPm, string postingUser,string content) :  isPm(isPm), postingUser(postingUser), content(content), op(op){}

string NotificationPacket::getIsPm (){
    if(this->isPm=='\0'){
        return "PM";
    }
   else return "Public";
}
string NotificationPacket::getPostingUser(){
    return this->postingUser;
}
string NotificationPacket::getContent(){
    return this->content;
}
short NotificationPacket::getOP() {
    return this->op;
}